### PR TITLE
Allow changing the format of the contents of the email

### PR DESCRIPTION
**Pull request by JeffersonBledsoe**
Allow the email that is sent to either be styled as a list (current behaviour) or as a table (new behaviour from this PR).

## Markup of emails

List:

```html
<p>A new form has been submitted from <strong>Form</strong>:</p>
  <ul>
    
      <li>
          <strong>Field 1:</strong> List
      </li>
    
  </ul>

```

Table:

```html
 <table>
    <caption>Email results for Form</caption>
    <thead>
      <tr role="row">
        <th scope="col" role="columnheader">Field</th>
        <th scope="col" role="columnheader">Value</th>
      </tr>
    </thead>
    <tbody>
      
        <tr role="row">
          <th scope="row" role="rowheader">Field 1</th>
          <td>Table</td>
        </tr>
      
    </tbody>
  </table>
```

### CODE REVIEW

File: src/collective/volto/formsupport/browser/send_mail_template_table.pt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<tal:root xmlns:i18n="http://xml.zope.org/namespaces/i18n"
+  i18n:domain="collective.volto.formsupport"
+  define="parameters python:options.get('parameters', {});
+            url python:options.get('url', '');
+            title python:options.get('title', '');">
+  <table>
+    <caption i18n:translate="send_mail_text">Email results for ${title}</caption>
```
Comment (cekk) [inline src/collective/volto/formsupport/browser/send_mail_template_table.pt:7]: If i remember correctly, this should be write like this:

<caption i18n:translate="send_mail_text">Email results for <tal:title i18n:name="title">${title}</tal:title>

And then when you update locales, title will be dynamic

File: src/collective/volto/formsupport/restapi/services/submit_form/post.py [hunk 2]
```diff
@@ -276,8 +282,25 @@ def send_data(self):
             self.send_mail(msg=msg, charset=charset)
 
     def prepare_message(self):
+        template_name = self.email_format_page_template_mapping[
+            self.block.get("email_format", "")
```
Comment (cekk) [inline src/collective/volto/formsupport/restapi/services/submit_form/post.py:286]: probably better like this:

```
email_format = self.block.get("email_format", "")
template_name = self.email_format_page_template_mapping.get(email_format, "send_mail_template")
```

So you don't need to set a "" key in mapping dict

File: src/collective/volto/formsupport/restapi/services/submit_form/post.py [hunk 3]
```diff
@@ -276,8 +282,25 @@ def send_data(self):
...
+        ]
+
+        if not template_name:
```
Comment (cekk) [inline src/collective/volto/formsupport/restapi/services/submit_form/post.py:289]: This shouldn't be an option because we always have a default value

**Review by cekk — CHANGES_REQUESTED** [review]

**Comment by JeffersonBledsoe** [comment]
@cekk I made a few of the changes you brought up. Not sure if I should update the locales or if that's to be done as part of a release.

File: src/collective/volto/formsupport/restapi/services/submit_form/post.py (hunk 3 above)
Comment (cekk) [reply src/collective/volto/formsupport/restapi/services/submit_form/post.py:288]: I mean something like this:

```suggestion
 email_format_page_template_mapping = {
        "list": "send_mail_template",
        "table": "send_mail_template_table",
    }
 
 ...
 
 email_format = self.block.get("email_format", "")
 template_name = self.email_format_page_template_mapping.get(email_format, 'send_mail_template')
 
```

In this case you always have a default value (send_mail_template)

**Comment by cekk** [comment]
> @cekk I made a few of the changes you brought up. Not sure if I should update the locales or if that's to be done as part of a release.

Yes, feel free to update locales to add new translations.

**Comment by JeffersonBledsoe** [comment]
@cekk Thanks for checking over this again. I've updated the locales now too

**Comment by cekk** [comment]
yes, will release with all the changes

**Pull request merged**